### ISSUE TITLE
V23: Cannot add a custom VaadinServlet in plain servlet starter 

### ISSUE THREAD

**Issue by taefi** [vaadin/flow#14288]
### Description of the bug
Tried the V23 skeleton-starter-flow by adding a custom VaadinServlet class and it crashes the application in a way that requesting http://localhost:8080/ would result in listing root directory contents instead of loading the `MainView`. It seems that adding "/VAADIN/*" is a must with the latest versions, but the other custom path such as `"/vaadin/*"` or `"/asdf/*"` is arbitrary. Basically, the presence of any custom `VaadinServlet` is enough to break the application's normal behavior. Adding a `HttpServlet` is fine and works as expected. 

### Minimal reproducible example
1. Download https://github.com/vaadin/skeleton-starter-flow/archive/v23.zip and unzip
2. Add a custom VaadinServlet class:
```java
import com.vaadin.flow.server.VaadinServlet;
import javax.servlet.annotation.WebServlet;

@WebServlet(asyncSupported = true, urlPatterns = { "/vaadin/*", "/VAADIN/*"})
public class WebComponentVaadinServlet extends VaadinServlet {
}
```
3. Run: `mvn clean jetty:run`
4. Hit the http://localhost:8080/

### Expected behavior
To see the `MainView` loaded.

### Actual behavior
It lists root directory contents:
![Screenshot 2022-08-08 at 13 45 41](https://user-images.githubusercontent.com/70698630/183400926-cb0af3e8-c68d-474f-bfcc-e97feebeb58c.png)

### Versions:
    - Vaadin / Flow version: Platform v23.1.4 and v23.2.0.alpha3  
    - Java version: 17
    - OS version: MacOs
    - Browser version (if applicable): Firefox 102.0.1 (64-bit), Chrome 103.0.5060.134 (Official Build) (x86_64)
    - Application Server (if applicable): Jetty
    - IDE (if applicable): N/A

**Comment by mcollovati** [comment]
If we define a custom servlet with a mapping different from root, I think it is expected not to see the main view at http://localhost:8080/.
I expect to see it for example at http://localhost:8080/asdf/ if mapping is `/asdf/*, /VAADIN/*`
And in this case it works for me

**Comment by taefi** [comment]
@mcollovati that's correct, but shouldn't it prevent the directory listing somehow?

**Comment by mcollovati** [comment]
Good point. So, you propose to somehow update jetty maven plugin configuration to prevent directory listing?

**Comment by taefi** [comment]
Not sure, but what I was expecting is that a servlet container either would match a servlet based on the requested URL and the registered servlets or not, and when it cannot find a proper handler it should show a 404 default error page. I don't think a Vaadin application should rely on a Maven/Gradle plugin to prevent directory listing.

**Comment by mcollovati** [comment]
My opinion is that if a user configures Vaadin to work on a mapping different from root, Vaadin should do nothing to prevent listing.
With jetty (but this holds for other containers) it is the default servlet that handles listing
So if the Vaadin servlet is not mapped to root, IMO it is up to the user to configure the servlet container to avoid listing.
